Blog Series 2: Blockchains (#34)
MAESTRO-Pool/website: content/blog/series_2_blog_2/index.md

# Cardano’s Unique Approach to the Blockchain

Cardano implements a unique approach to tackling the blockchain. Cardano utilizes rigorous mathematical provability, novel methods such as eUTxOs, and layering of computation and transactions to build a fast and secure foundation to build arbitrarily complex systems on top of, and drive the next generation of connectivity.

> Cardano utilizes a version of the Proof-of-Stake [LINK] model to dictate how they come to consensus when adding information to the blockchain. This protocol is called Ouroboros [LINK]. It has some unique properties that make it very special. Most importantly, it is that the protocol has provable security[https://hydra.iohk.io/build/6684352/download/1/network-design.pdf]. This means that, much like a mathematical equation, we can consistently prove, given some set of conditions, we will always maintain a specific level of security.

Reviewer: ```suggestion
Cardano utilizes a version of the Proof-of-Stake [LINK] model to dictate how they come to consensus when adding information to the blockchain. This protocol is called Ouroboros [LINK]. It has some unique properties that make it very special. Most importantly, it is that the protocol has provable security[https://hydra.iohk.io/build/6684352/download/1/network-design.pdf]. This means that, much like a mathematical equation, we can consistently prove that given a set of conditions, we will always maintain a specific level of security.
```

> These security guarantees come from a strong foundation however. We assume that because the owners of Cardano tokens (Stakeholders of ADA) generally want to hold as much as they can, they have an honest majority. By holding a lot of Cardano, the Ouroboros model assumes that they have a vested interest in making the network succeed for as long as they can. They don’t want it to fail, or be attacked by malicious users, otherwise their token value will go down, there will be less stakeholders, which in turn will reduce the network security. So, there is a great deal of faith, that honest majority holds, and that smaller malicious actors cannot manipulate the system, because the majority is incentivized to keep the network intact, and don’t plan on destroying the network they have such a majority stake in.

Reviewer: ```suggestion
These security guarantees come from a strong foundation, however. We assume that because the owners of Cardano tokens (Stakeholders of ADA) generally want to hold as much as they can, they have an honest majority. By holding a lot of Cardano, the Ouroboros model assumes that they have a vested interest in making the network succeed for as long as they can. They don’t want it to fail or be attacked by malicious users. This will cause the token value to go down, decrease the number of stake holders and reduce the overal network security. Thus, there is a great deal of faith that the honest majority holds and that smaller malicious actors cannot manipulate the system. The majority of stakeholders are incentivized to keep the network intact and prevent it from being destroyed.
```

![Image of UTxO and Accounting Based from Horizen Academy](utxo_account_based_display.png)

## Cardano's Extended UTxO Model 

When deciding between UTxO and Account based models, you need to account for an important component that affects Cardano greatly! This is the idea of smart contracts. We don’t get into the nitty gritty but smart contracts are essentially code that is written into the blockchain to make decisions on how transactions are made for a specific user, and encode logically decisions that any regular program would have, except, for transactions and really any interaction on the blockchain.

In our case, an account-based model greatly benefits us, because it provides the intuitive modes of calculating whether we do a certain action, and lets us easily decide over which account we send or expect to receive a currency from. 

> The main issue with account-based, is that you need to create specific accounts/wallets for the contracts, and the contract logic that needs to be used, must encompass all possible edge cases to avoid being exploited. For example, with Ethereum, users must define a contract for a user, and every transaction that is given to this user must be fully encompassing, move value around from all the required wallets, interact with other smart contract accounts, and then generate the correct result. And once it's published to the blockchain, everyone can interact with it. There are countless examples [link examples] of exploits being taken advantage of, or poor design, which ends up losing thousands of Etheruem to exploits and attacks.

Reviewer: The main issue with account-based, is... => The main issue with account-based is...

Links to examples?

Reviewer: ```suggestion
The main issue with account-based is that you need to create specific accounts/wallets for the contracts, and the contract logic that needs to be used, must encompass all possible edge cases to avoid being exploited. For example, with Ethereum, users must define a contract for a user, and every transaction that is given to this user must be fully encompassing, move value around from all the required wallets, interact with other smart contract accounts, and then generate the correct result. And once it's published to the blockchain, everyone can interact with it. There are countless examples [link examples] of exploits being taken advantage of, or poor design, which ends up losing thousands of Etheruem to exploits and attacks.
```

> In the case of Cardano, they chose the route of extending the UTxO model, by allowing for Transactions themselves to contain smart contract code, while also enforcing that any interaction with this newly modified UTxO, also uses the same contract code. By adding custom data to outputs (in addition to value), and by allowing for more "locks" and “keys” deciding under which condition an output can be unlocked for consumption by a transaction. In other words, instead of just having public keys (hashes) for locks and corresponding signatures serving as “keys”, EUTXO enables arbitrary logic in the form of scripts. This arbitrary logic inspects the transaction and the data to decide whether the transaction is allowed to use an input or not.

Reviewer: In the case of Cardano, they chose the route of extending the UTxO model, by allowing for Transactions themselves to contain smart contract code, while also enforcing that any interaction with this newly modified UTxO, also uses the same contract code.
=>
In the case of Cardano, they chose the route of extending the UTxO model, by allowing for Transactions themselves to contain smart contract code, while also enforcing that any interaction with this newly modified UTxO also uses the same contract code.

Reviewer: ```suggestion
Cardano chose the route of extending the UTxO model, by allowing for transactions themselves to contain smart contract code, while also enforcing that any interaction with this newly modified UTxO also uses the same contract code. By adding custom data to outputs (in addition to value), and by allowing for more "locks" and “keys” deciding under which condition an output can be unlocked for consumption by a transaction. In other words, instead of just having public keys (hashes) for locks and corresponding signatures serving as “keys”, EUTXO enables arbitrary logic in the form of scripts. This arbitrary logic inspects the transaction and the data to decide whether the transaction is allowed to use an input or not.
```

> With this, the nodes validating the network can be highly efficient processing transactions in parallel, and allow for many composable interactions between other smart contracts, without as much risk. 

Reviewer: ```suggestion
With this, the nodes validating the network can efficiently process transactions in parallel, and allow for many composable interactions between other smart contracts, without as much risk! 
```

## Settling Layer & Computational Layer

> When deciding when to utilize which model works best for dealing with transactions, Cardano took the best of both worlds by creating an Extended UTxO model. Account-based model provides great utility when attempting to manage the computational layer that is built on top of the consensus model and UTxO provides a great way of dealing the scalability issue of tracking different tokens as they move throughout the network and managing them from each of the [stake pools in the slot](LINK).  

Reviewer: ```suggestion
Cardano took the best of both worlds by creating the **Extended UTxO** model. The account-based model provides great utility when attempting to manage the computational layer that is built on top of the consensus model. The UTxO model provides a great way of dealing with the scalability issue of tracking different tokens as they move throughout the network and managing them from each of the [stake pools in the slot](LINK).  
```

[Pretty picture here]

> Cardano then utilizes the EUTxO model in their computational layer, which is in all intent and purpose, is almost like a second block chain, and UTxO for the settlement layer, or the place where they want to handle actually moving the core token around. The reason being, that, with previous cryptocurrencies like Ethereum, you were forced into their account based model, where in order to use the smart contract, you had converse with its base token (ether or ETH), which meant your tokens would always be in the language of eth, and would have to deal with the problems of Ether [5].

Reviewer: ```suggestion
Cardano utilizes the EUTxO model for the computational layer, which is almost like a second block chain, and the UTxO model for the settlement layer, where they handle the transfer of tokens. With previous cryptocurrencies like Ethereum, you were forced into their account based model, which required that your tokens be based on ETH[5].

```

> On the other hand, with Cardano, if you want to settle transactions, you can chose to, and if you want to compute, you can interact strictly with the computational layer, and all of the network validators, can chose to validate for which ever they prefer, or even both! Combined with native multi-asset (essentially create your own coins to keep track of) you could compute and settle independently of Cardano’s main token (ADA)! This approach greatly solves the scaling problems we see with Ethereum [link scaling article].

Reviewer: ```suggestion
With Cardano, you can either settle transactions or handle compute. All of the network validators can validate either or both! Combined with native multi-asset (essentially create your own coins to keep track of) you could compute and settle independently of Cardano’s main token (ADA)! This approach greatly solves the scaling problems we see with Ethereum [link scaling article].
```

> ## Conclusion 

Reviewer: Given the complexity of what's presented in this blog post, a solid and punchy Conlcusion is critical here.
